src/node_report_module.cc

```diff
@@ -0,0 +1,315 @@
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+
+  // Trigger report if requested
+  if (options->report_catch_exception) {
+    TriggerNodeReport(isolate, env, "exception", __func__, nullptr, info[0]);
+  }
+}
+
+/*******************************************************************************
+ * Signal handler for report action, called from JS land (util.js)
+ ******************************************************************************/
+
+void OnUserSignal(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  Isolate* isolate = env->isolate();
+  CHECK(info[0]->IsString());
+  Local<String> str = info[0].As<String>();
+  String::Utf8Value value(isolate, str);
+  TriggerNodeReport(
+      isolate,
+      env,
+      *value,
+      __func__,
+      nullptr,
+      MaybeLocal<Value>());
+}
+
+/*******************************************************************************
+ * Native module initializer function, called when the module is require'd
+ ******************************************************************************/
+void InitializeNodeReport(Isolate* isolate, Environment* env) {
+  // Register the boot time of the process, for
+  // computing resource consumption average etc.
+  // std::string options;
+  IsolateData* id = env->isolate_data();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+  time(&boot_time);
+  if (options->report_catch_fatalerror && id->error_hook_initialised == false) {
+    env->isolate()->SetFatalErrorHandler(OnFatalError);
+    id->error_hook_initialised = true;
+  }
+
+  if (options->report_signal == "") options->report_signal = "SIGUSR2";
+}
+// A method to sync up data elements in the JS land with its
+// corresponding elements in the C++ world. Required because
+// (i) the tunables are first intercepted through the CLI but
+// later modified via APIs. (ii) the report generation events
+// are controlled partly from C++ and partly from JS.
+void SyncConfig(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  Local<Context> context = env->context();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+  IsolateData* id = env->isolate_data();
+
+  CHECK_EQ(info.Length(), 2);
+  Local<Object> obj = info[0]->ToObject(context).ToLocalChecked();
+  bool sync = info[1].As<Boolean>()->Value();
+
+  // events array
+  Local<String> eventskey = FIXED_ONE_BYTE_STRING(env->isolate(), "events");
+  Local<Value> events_unchecked =
+      obj->Get(context, eventskey).ToLocalChecked();
+  Local<Array> events;
+  if (events_unchecked->IsUndefined() || events_unchecked->IsNull()) {
+    events_unchecked = Array::New(env->isolate(), 0);
+    obj->Set(context, eventskey, events_unchecked);
+  }
+  events = events_unchecked.As<Array>();
+
+  // signal
+  Local<String> signalkey = FIXED_ONE_BYTE_STRING(env->isolate(), "signal");
```

Comment:
`env->signal_string()` works too, and is more performant if called multiple times because it only creates a single string instance

Reply:
done, changed to `env->signal_string()`